pipelines/matrix/src/matrix/cli_commands/run.py

```diff
@@ -74,8 +75,9 @@ class RunConfig(NamedTuple):
 @click.option( "--conf-source",   type=click.Path(exists=True, file_okay=False, resolve_path=True), help=CONF_SOURCE_HELP,)
 @click.option( "--params",        type=click.UNPROCESSED, default="", help=PARAMS_ARG_HELP, callback=_split_params,)
 @click.option( "--from-env",      type=str, default=None, help="Custom env to read from, if specified will read from the `--from-env` and write to the `--env`",)
+@click.option( "--from-run",      type=str, default=None, help="Run to read from, if specified will read from the `--from-run` datasets and write to the run_name datasets",)
```

Comment:
Think we should try to document how this works somewhere, as it's pretty custom. Maybe the diagrams we did?